assessments/projects/openfga/self-assessment.md

```diff
@@ -0,0 +1,277 @@
+- Storing and retrieving and authorization models.
+- Evaluating the inferred permissions for a given subject and object.
+
+**Database Server**
+
+Stores the relationship tuples and authorization models, as well as a changelog. Currently support Postgres and MySQL.
+
+**CLI/API Clients**
+
+Make API requests to the OpenFGA server, i.e. creating/querying relationship tuples, updating authorization models, checking for access, or listing objects a user has access to. 
+
+Clients can use either no authentication, shared key, or OAuth client credentials as a method for authentication.
+
+### Actions
+
+**Invoking the OpenFGA APIs**
+
+Every time a server endpoint is invoked, OpenFGA validates that:
+
+  - The credentials provided in the API call match the ones configured in the server.
+
+  - Ensures that the input is a semantically valid, e.g. that a tuple is valid according to the authorization model or that the model does not have disallowed cyclical or problematic definitions
+
+  - Payload Verification: 
+  
+    - Confirm that API payloads adhere to Protobuf API definitions.  
+    - Validate parameters for proper structure, e.g. ensuring users are written in the correct format which is 'user:<userid>'
+  
+**Writing an Authorization Model**
+
+  - Semantic Model Verification: OpenFGA validates that Authorization Models are semantically valid, avoiding cyclical or problematic definitions and other disallowed criteria.
+
+  - Version and Configuration: When writing a model, a new version is created (models are immutable). Applications must be configured to use the new version after validation and confirmation of expected behavior.
+
+**Calling the Authorization Query endpoints**
+
+When the [/check](https://openfga.dev/api/service#/Relationship%20Queries/Check) and [/list-objects](https://openfga.dev/api/service#/Relationship%20Queries/ListObjects) endpoints are called, OpenFGA limits the number of simultaneous paths explored and enforces depth limitations on the graph traversal. 
+
+To protect against DoS attacks, OpenFGA restricts both the number of simultaneous paths explored and the depth of paths traversed in the graph.
+
+**Upgrading OpenFGA Database Schema**
+
+Database Migration Planning: When installing new versions of OpenFGA, database migrations can be executed in a way that minimizes downtime and ensure a smooth transition in the target system.
+
+### Goals
+
+* Simplify and standardize authorization processes, making them more consistent across various applications and systems.
+
+* Establish patterns and standards for externalized authorization.
+
+* Create architectural patterns, terminologies, and protocols that enable interoperability among different authorization systems.
+
+* Deliver an authorization service for any application component.
+
+* Enable centralized authorization decisions and permits diverse teams to implement authorization using a shared framework across various application components.
+
+### Non-Goals

```

Comment:
I'm putting my comments for future reference when the team is considering syncing with OpenFGA security self-assessment. The non-goals and goals need to updated it is to generic. Some parts of goals and non-goals are overlapping.